src/libNode/Node.cpp

```diff
@@ -446,32 +447,57 @@ bool Node::CheckIntegrity(bool fromIsolatedBinary) {
       if (BlockStorage::GetBlockStorage().GetMicroBlock(mbInfo.m_microBlockHash,
                                                         mbptr)) {
         if (LOOKUP_NODE_MODE) {
-          auto tranHashes = mbptr->GetTranHashes();
+          const auto& tranHashes = mbptr->GetTranHashes();
           for (const auto& tranHash : tranHashes) {
             TxBodySharedPtr tx;
             if (!BlockStorage::GetBlockStorage().CheckTxBody(tranHash)) {
-              LOG_GENERAL(WARNING, "Missing Tx: " << tranHash);
-              result = false;
+              LOG_GENERAL(WARNING, "FB: " << txBlock->GetHeader().GetBlockNum()
+                                          << " MB: " << mbInfo.m_shardId
+                                          << " Missing Tx: " << tranHash);
+              *result = false;
               if (!fromIsolatedBinary) {
                 break;
               }
             }
           }
         }
       } else {
-        LOG_GENERAL(WARNING, "Missing MB: " << mbInfo.m_microBlockHash);
-        result = false;
+        LOG_GENERAL(WARNING,
+                    "FB: " << txBlock->GetHeader().GetBlockNum()
+                           << " Missing MB: " << mbInfo.m_microBlockHash);
+        *result = false;
         if (!fromIsolatedBinary) {
           break;
         }
       }
     }
-    if (!result && !fromIsolatedBinary) {
+  };
+
+  const unsigned int NUMTHREADS = 10;
+  const int MAXJOBSLEFT = NUMTHREADS * 3;
```

Comment:
Do not understand this 3 here? Is it arbitrary?

Reply:
Yep, just arbitrarily defined the backlog limit to 3 times the number of threads